application/classes/Ushahidi/Rest.php

```diff
@@ -266,7 +266,9 @@ protected function _check_access()
 		}
 		catch (OAuth2Exception $e)
 		{
-			if (!$this->_is_auth_required() AND $e instanceof MissingAccessTokenException)
+			if (!$this->_is_auth_required() AND (
+				$e instanceof MissingAccessTokenException OR
+				$e instanceof InvalidAccessTokenException))
```

Comment:
I'm not sure about this change TBH. Its important we fail invalid tokens, so that the client knows the token isn't valid. Rather than just try to run them as unauth'd.